File: app/src/main/java/dev/trindadedev/anims/MainActivity.java
package dev.trindadedev.anims;

import android.graphics.BitmapFactory;
import android.graphics.Point;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import dev.trindadedev.anims.databinding.ActivityMainBinding;
import dev.trindadedev.anims.ui.components.DragView;

public class MainActivity extends AppCompatActivity {
  private ActivityMainBinding binding;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    binding = ActivityMainBinding.inflate(getLayoutInflater());
    setContentView(binding.getRoot());

    binding.dragView.setSize(200, 200);
    binding.dragView.setPosition(new Point(50, 50));
  }

  @Override
  protected void onDestroy() {
    super.onDestroy();
    this.binding = null;
  }
}

File: app/src/main/java/dev/trindadedev/anims/MainActivity.kt
package dev.trindadedev.anims

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import dev.trindadedev.anims.databinding.ActivityMainBinding

public class MainActivity : AppCompatActivity() {
  
  private var _binding: ActivityMainBinding? = null
  private val binding: ActivityMainBinding
    get() = checkNotNull(_binding) { "Activity has been destroyed" }
  
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    _binding = ActivityMainBinding.inflate(layoutInflater)
    setContentView(binding.root)
  }
  
  override fun onDestroy() {
    super.onDestroy()
    _binding = null
  }
}
